src/js/experiments/snippetFeedback.js

```diff
@@ -0,0 +1,188 @@
+   * */
+  renderFeedbackForm() {
+    if (!this.feedbackForm) {
+      const container = this.makeElement({ kind: 'div', className: 'form' });
+      const prompt = this.makeElement({
+        kind: 'div',
+        text: 'Any way we can improve?',
+      });
+
+      const textForm = this.makeElement({
+        kind: 'textarea',
+      });
+
+      const sendButton = this.makeElement({
+        kind: 'button',
+        className: 'text-btn',
+        text: 'Send',
+        onClick: () => {
+          this.trackFormSubmission(textForm.value);
+          this.wasThisHelpfulContainer.innerHTML =
+            '<div>Thank you for your feedback!</div>';
+        },
+      });
+
+      container.appendChild(prompt);
+      container.appendChild(textForm);
+      container.appendChild(sendButton);
+      this.feedbackForm = container;
+      this.wasThisHelpfulContainer.appendChild(container);
+    }
+  }
+
+  // -- Trackers --
+  // These are responsible for sending data based on user input
+  // to amplitude.
+
+  /**
+   * trackYesOrNoButton sends a trackAction to aplitude.
+   * actions are: "docs-snippet-helpful-no"  and "docs-snippet-helpful-yes"
+   * */
+  trackYesOrNoButton(yesOrNoString) {
+    window.AnalyticsClient.trackAction(
+      `docs-snippet-helpful-${yesOrNoString}`,
+      {
+        originatingSnippet: this.snippetElement.textContent,
+        wasHelpful: yesOrNoString,
+        timeOfButtonClick: Date.now(),
+      },
+    );
+  }
+
+  /**
+   * `trackFormSubmission` invokes trackAction with a payload of the original
+   * snippet, the feedback of the user, and the timeOfFormSubmission.
+   * */
+  trackFormSubmission(formContent) {
+    window.AnalyticsClient.trackAction(`docs-snippet-helpful-form-submission`, {
+      originatingSnippet: this.snippetElement.textContent,
+      feedback: formContent,
+      wasThisHelpful: this.wasThisHelpful,
+      // for diffing this against the time that the trackYesOrNoButton was clicked
+      // as well as possible the time the copy code button was clicked (TODO add that)
+      timeOfFormSubmission: Date.now(),
+    });
+  }
+
+  // -- Helpers --
+
+  /**
+   * Construct a dom element, setting classes, text content, class, onclick etc.
+   * */
+  makeElement({ kind, text, className, onClick }) {
+    let el = document.createElement(kind);
+    if (text) el.textContent = text;
+    if (className) el.classList.add(className);
+    if (onClick) el.addEventListener('click', onClick, { once: true });
+
+    return el;
+  }
+}
+
+/**
+ * init queries the dom for all snipppets
+ * TODO(blocked): when that snippet's COPY CODE button is clicked... (waiting for graduation of copy code buttons)
+ *
+ * */
+function init() {
+  // add event listeners to all code snippets on page
+  window.OptimizelyClient.getVariationName({
+    experimentKey: 'dd_docs_snippet_feedback_experiment_test',
+    groupExperimentName: 'q4_fy22_docs_disco_experiment_group_test',
+    experimentContainer: '.hljs',
```

Comment:
can we add a link to the experiment to optimizely here?

Reply:
Done!